File: tests/parsers/fixtures/sample.go
// Package sample provides example types for parser testing.
package sample

import (
	"fmt"
	"strings"
)

// DataProcessor handles data transformation.
type DataProcessor struct {
	Config map[string]string
}

// NewDataProcessor creates a new DataProcessor instance.
func NewDataProcessor(config map[string]string) *DataProcessor {
	return &DataProcessor{Config: config}
}

// Process transforms a slice of strings.
func (dp *DataProcessor) Process(data []string) []string {
	result := make([]string, 0, len(data))
	for _, item := range data {
		result = append(result, dp.transform(item))
	}
	return result
}

func (dp *DataProcessor) transform(item string) string {
	return strings.TrimSpace(fmt.Sprintf("%s_processed", item))
}

File: tests/parsers/fixtures/sample.php
<?php

namespace App\Sample;

use App\Contracts\ProcessorInterface;
use App\Support\Helper;

/**
 * Sample PHP class for parser testing.
 */
class DataProcessor implements ProcessorInterface
{
    private string $config;

    public function __construct(string $config)
    {
        $this->config = $config;
    }

    public function process(array $data): array
    {
        return array_map([$this, 'transform'], $data);
    }

    private function transform(string $item): string
    {
        return strtolower(trim($item));
    }
}
